### DISCUSSION TITLE
Build system sourced properties (e.g. Maven) are no longer available at runtime

### DISCUSSION THREAD

**Discussion by anchialas** [quarkusio/quarkus#44935] [Q&A]
For reasons of convenience (and because it worked with Quarkus 3.8 LTS) we have set various Quarkus configuration properties as Maven properties in our Maven parent pom, e.g. `quarkus.log.console.format` or `quarkus.analytics.disabled`.

With the upgrade to 3.15 LTS this unfortunately no longer works, specifically the `quarkus.log.console.format` is no longer adopted. With PR [39604 ](https://github.com/quarkusio/quarkus/pull/39604) a clear separation of build-time and runtime properties was introduced. The necessary config source “Build system” was removed with [1]. 

This change is striking (for us) and went unnoticed for the time being, probably also because this behavior was not so clearly documented. My question is whether there is an easy way with v3.15+ to continue to have Maven properties as runtime config source?

I would appreciate any help.
Kind regards

[1] https://github.com/quarkusio/quarkus/pull/39604/files#diff-1f9bfdc8e0745811c96617eca0fbb93ebc51b44e98ae4c6906bfa5d5700ce2a5R1151

[a comment by an automated account is not shown]

**Reply by aloubyansky** [reply]
FYI @radcortez

**Comment by radcortez** [comment]
> My question is whether there is an easy way with v3.15+ to continue to have Maven properties as runtime config source?

Unfortunately, no. Due to https://github.com/quarkusio/quarkus/issues/39388, we think it is too dangerous to let some build-time sources contribute to runtime. We can argue that `quarkus.log.console.format` or `quarkus.analytics.disabled` are harmless compared to the hibernate example, but we don't really have a good way to know which configuration is safe or not.

**Comment by anchialas** [comment]
Thank you for your fast feedback. I see the problem and understand the decision.

Our solution: We now declared the properties as application properties in a library.

**Reply by radcortez** [reply]
Thank you for your understanding :)

Yes, that would work :)
